Call to register_engine to add ourselves to the extname_map

ngan opened the pull request:
I'm using the `jasmine-rails` and I'm getting this error:

```
couldn't find file 'foo_spec.js.es6' with type 'application/javascript'
```

After much debugging, it turns out es6 isn't even in the ext graph that sprocket generates.  This PR adds es6 and fixes the issue.

ngan committed:
Call to register_engine to add ourselves to the extname_map

TannerRogalsky commented:
@ngan Is it possible that you're including the `.es6` extension when you don't need to? We register the extension as a new mime type so that, generally, you should never need to have the extension in the filename when you reference it.

ngan commented:
Hm, if I don't have the `.es6` extension then the file won't be processed/converted.  It'll just render ES6 JS and will cause syntax errors in the browser.

TannerRogalsky commented:
Can you replicate this behaviour in a test case? The behaviour that you're describing is outside of my experience so, while your code makes sense to me, I'd like to understand the why of the matter before I merge it.

ngan commented:
Hi @TannerRogalsky I've updated the PR with tests...they probably don't look great.

But what I did find out what that jasmine-spec was using `.js.erb` and using the `require_asset` method:
https://github.com/searls/jasmine-rails/blob/master/lib/assets/javascripts/jasmine-specs.js.erb

Also, if you look at my changes, I added a file called `test/fixtures/file.js.es6`.  I'm not sure why I need `.js.es6` for it to work.  `.es6` by itself wouldn't work.

ngan commented:
I'm also a bit tad bit confused on how react-rails got theirs to work (if it does):
https://github.com/reactjs/react-rails/commit/c8ecba2fd799b72ad77dd926bb814eff498b9431

TannerRogalsky commented:
@ngan I appreciate your diligence here. I'm still not totally sure why Sprockets isn't being consistent about this but I agree that this seems to resolve this sort of weird issue. I'm going to leave this open for ~24 hours (incase anyone has any insight/objections) and then merge it and bump the gem.

Thanks for the PR!

TannerRogalsky commented:
These changes were published in v0.8.2.

ngan commented:
Thanks @TannerRogalsky :clap:
